store/src/java/com/zimbra/cs/event/SolrEventDocument.java

```diff
@@ -37,41 +36,48 @@ public SolrInputDocument getDocument() {
         return document;
     }
 
-    private String generateId() {
+    public Object[] getIdParts() {
         UniqueOn uniqueOn = event.getEventType().getUniqueOn();
         String eventType = event.getEventType().name();
+        String dsId = event.getDataSourceId();
         Integer msgId;
-        String accountLevelIdentifier;
+        Object[] idParts;
         switch (uniqueOn) {
         case ACCOUNT:
-            accountLevelIdentifier = event.getEventType().name();
+            idParts = new Object[] {eventType};
             break;
         case DATASOURCE:
-            accountLevelIdentifier = String.format("%s:%s", eventType, event.getDataSourceId());
+            idParts = new Object[] {eventType, dsId};
             break;
         case MESSAGE:
             msgId = (Integer) event.getContextField(EventContextField.MSG_ID);
-            accountLevelIdentifier = String.format("%s:%d", eventType, msgId);
+            idParts = new Object[] {eventType, msgId};
             break;
         case MSG_AND_SENDER:
             msgId = (Integer) event.getContextField(EventContextField.MSG_ID);
             String sender = (String) event.getContextField(EventContextField.SENDER);
```

Comment:
I presume 'getContextField' doesn't have a method to return a default value instead of null?